src/unix/tty.c

```diff
@@ -51,6 +54,18 @@ int uv_tty_init(uv_loop_t* loop, uv_tty_t* tty, int fd, int readable) {
   flags = 0;
   newfd = -1;
 
+  /* Try to find a tty in stdin, stdout, and stderr. If a tty is found compare
+   * it to the fd. Only reopen the file descriptor if its inode matches the one
+   * bound to stdin, stdout, or stderr
+   */
+  for(fdtty = 0; !isatty(fdtty) && fdtty < 3; fdtty++);
```

Comment:
ISTM that the second clause should be `fdtty < 3 && !isatty(fdtty)` to avoid an unnecessary `isatty()` call.